stko/calculators/orca_calculators.py

```diff
@@ -0,0 +1,352 @@
+    When running :meth:`calculate`, this calculator changes the
+    present working directory with :func:`os.chdir`. The original
+    working directory will be restored even if an error is raised, so
+    unless multi-threading is being used this implementation detail
+    should not matter.
+
+    If multi-threading is being used an error could occur if two
+    different threads need to know about the current working directory
+    as :class:`.OrcaEnergy` can change it from under them.
+
+    Note that this does not have any impact on multi-processing,
+    which should always be safe.
+
+    Examples
+    --------
+
+    .. code-block:: python
+
+        import stk
+        import stko
+
+        bb1 = stk.BuildingBlock('NCCNCCN', [stk.PrimaryAminoFactory()])
+        bb2 = stk.BuildingBlock('O=CCCC=O', [stk.AldehydeFactory()])
+        polymer = stk.ConstructedMolecule(
+            stk.polymer.Linear(
+                building_blocks=(bb1, bb2),
+                repeating_unit="AB",
+                orientations=[0, 0],
+                num_repeating_units=1
+            )
+        )
+
+        # Optimize the constructed molecule so that it has a
+        # reasonable structure.
+        opt = stko.UFF()
+        polymer = opt.optimize(polymer)
+
+        # Calculate energy using GFN-xTB.
+        orca = stko.OrcaEnergy(
+            orca_path='/opt/orca/orca',
+            topline='! SP B97-3c',
+        )
+
+        orca_results = orca.get_results(polymer)
+
+        # Extract properties from the energy calculator for a given
+        # molecule.
+        total_energy = orca_results.get_total_energy()
+
+    If you plan on running calculations on a cluster, outside a python
+    environment, with very specific settings :mod:`stko` does not apply
+    or just want the input file written, you can use the
+    `write_input_only` argument to save the input file in the
+    `output_dir` as `orca_input.inp` with the input xyz file as
+    `input_structure.xyz`.
+
+    .. code-block:: python
+
+        # Optimize the constructed molecule so that it has a
+        # reasonable structure.
+        optimizer = stko.ETKDG()
+        polymer = optimizer.optimize(polymer)
+
+        # Calculate energy using GFN-xTB.
+        orca = stko.OrcaEnergy(
+            orca_path='/opt/orca/orca',
+            topline='! SP B97-3c',
+            write_input_only=True,
+        )
+
+        orca.get_results(polymer)
+
+    References
+    ----------
+    .. [1] https://orcaforum.kofo.mpg.de/app.php/portal
+
+    """
+    def __init__(
+        self,
+        orca_path,
+        topline,
+        basename=None,
+        output_dir=None,
+        num_cores=1,
+        charge=0,
+        multiplicity=1,
+        write_input_only=False,
+    ):
+        """
+        Initializes a :class:`OrcaEnergy` instance.
+
+        Parameters
+        ----------
+        orca_path : :class:`str`
+            The path to the Orca executable.
+
+        topline : :class:`str`
```

Comment:
I think for DFT codes that is a bad idea because of:

1. way too many options
2. I would not use a dict as an input argument anyway, I would stick to the xtb format, which is a keyword for all changes, which is just too much! 

This is the most flexible approach I could think of.